internal/release/services/release/release.go

```diff
@@ -54,6 +72,170 @@ func (s *ReleaseService) RegisterWithServer(grpcServer grpc.ServiceRegistrar) er
 	return nil
 }
 
+// Publishes a new release.
+func (s *ReleaseService) Publish(ctx context.Context, request *release.PublishRequest) (*release.PublishResponse, error) {
+
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(
+		attribute.String("release.name", request.GetName()),
+		attribute.String("release.version", request.GetVersion()),
+	)
+
+	mv := Release{
+		Type:         request.GetType(),
+		Organization: request.GetOrganization(),
+		Name:         request.GetName(),
+		Version:      request.GetVersion(),
+		Description:  request.GetDescription(),
+		Links:        request.GetLinks(),
+		CreatedAt:    time.Now().UTC().String(),
+	}
+
+	av, err := attributevalue.MarshalMap(mv)
+
+	if err != nil {
+		span.RecordError(err)
+		return nil, MarshalReleaseError
+	}
+
+	in := &dynamodb.PutItemInput{
+		Item:      av,
+		TableName: aws.String(ReleaseTableName),
+	}
+
+	if _, err = s.Db.PutItem(ctx, in); err != nil {
+		span.RecordError(err)
+		return nil, PublishReleaseError
+	}
+
+	log.Println("New release created.")
+	return ReleasePublished, nil
+}
+
+// ListReleases Retrieves all releases.
+// Only releases that have been published should be reported.
+func (s *ReleaseService) ListReleases(ctx context.Context, request *releaseSvc.ListReleasesRequest) (*releaseSvc.ListReleasesResponse, error) {
+
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(
+		attribute.String("release.list", "Lists releases based on filters"),
```

Comment:
Replace this with the values in the `request` argument, ie I would expect to see MaxAgeSeconds, types, organizations and page as attributes.